addons/escoria-core/game/core-scripts/esc/commands/queue_event.gd

```diff
@@ -87,3 +95,27 @@ func run(arguments: Array) -> int:
 func interrupt():
 	# Do nothing
 	pass
+
+
+# Fetches the object node or current room containing the desired ESC script.
+#
+# PRE: If global_id represents a room, then `escoria.main.current_scene` must be valid.
+#
+# **Parameters**
+#
+# - global_id: ID of the object or room with the desired ESC script.
+#
+# *Returns*
+#
+# The object node or the current room.
+func _get_script_node(global_id: String):
+	var node = null
+
+	if escoria.object_manager.has(global_id):
+		node = escoria.object_manager.get_object(
+			global_id
+		).node
+	else:
+		node = escoria.main.current_scene
```

Comment:
In this code, we'll return the room node script when `global_id` doesn't refer to a registered item. This will work, but do we want a check for this ?

Reply:
I had thought about that, but given that we've already done that check prior to calling this method, I put a precondition ("PRE") in the docstring. I'm not a fan of duplicating code if possible. Happy to discuss, though.

Reply:
Nothing against using a PRE in docstring here, but I find it rather incomplete then: I'd add a notice whether provided global_id is null or inexisting would default to returning the room script.